bin/utils/logging.sh

```diff
@@ -0,0 +1,62 @@
+# bin/utils/logging.sh - Logging and output functions
+# Source this file to get standardized print functions for terminal output.
+#
+# Usage:
+#   source "$(dirname "${BASH_SOURCE[0]}")/utils/logging.sh"
+#   print_header "My Script"
+#   print_step "Doing something"
+#   print_success "It worked"
+#
+# Provides: print_header, print_step, print_success, print_error, print_warning, print_info
+
+# Guard against multiple inclusion
+[[ -n "${_DEVUTILS_LOGGING_LOADED:-}" ]] && return 0
+_DEVUTILS_LOGGING_LOADED=1
+
+# Source dependencies
+_DEVUTILS_LOGGING_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+source "$_DEVUTILS_LOGGING_DIR/colors.sh"
+unset _DEVUTILS_LOGGING_DIR
+
+# Print a prominent header (for script titles, major sections)
+print_header() {
+    echo -e "\n${BOLD}${BLUE}===================================================================${NC}"
+    echo -e "${BOLD}${BLUE}  $1${NC}"
+    echo -e "${BOLD}${BLUE}===================================================================${NC}\n"
+}
+
+# Print a section header (for grouping related output)
+print_section() {
+    echo -e "\n${BOLD}$1${NC}"
+    echo -e "${BOLD}$(printf -- '-%.0s' {1..63})${NC}"
```

Comment:
want to match the print_header function's length?

```suggestion
    echo -e "${BOLD}$(printf -- '-%.0s' {1..67})${NC}"
```